daemon/src/main/java/bisq/daemon/grpc/GrpcMoneroNodeService.java

```diff
@@ -0,0 +1,153 @@
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with Haveno. If not, see <http://www.gnu.org/licenses/>.
+ */
+package bisq.daemon.grpc;
+
+import bisq.core.api.CoreApi;
+import bisq.core.xmr.MoneroNodeSettings;
+
+import bisq.proto.grpc.GetMoneroNodeSettingsReply;
+import bisq.proto.grpc.GetMoneroNodeSettingsRequest;
+import bisq.proto.grpc.IsMoneroNodeRunningReply;
+import bisq.proto.grpc.IsMoneroNodeRunningRequest;
+import bisq.proto.grpc.MoneroNodeGrpc.MoneroNodeImplBase;
+import bisq.proto.grpc.StartMoneroNodeReply;
+import bisq.proto.grpc.StartMoneroNodeRequest;
+import bisq.proto.grpc.StopMoneroNodeReply;
+import bisq.proto.grpc.StopMoneroNodeRequest;
+
+import io.grpc.ServerInterceptor;
+import io.grpc.stub.StreamObserver;
+
+import javax.inject.Inject;
+
+import java.util.HashMap;
+import java.util.Optional;
+
+import lombok.extern.slf4j.Slf4j;
+
+import static bisq.daemon.grpc.interceptor.GrpcServiceRateMeteringConfig.getCustomRateMeteringInterceptor;
+import static bisq.proto.grpc.MoneroNodeGrpc.getStartMoneroNodeMethod;
+import static bisq.proto.grpc.MoneroNodeGrpc.getStopMoneroNodeMethod;
+import static bisq.proto.grpc.MoneroNodeGrpc.getIsMoneroNodeRunningMethod;
+import static bisq.proto.grpc.MoneroNodeGrpc.getGetMoneroNodeSettingsMethod;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+import bisq.daemon.grpc.interceptor.CallRateMeteringInterceptor;
+import bisq.daemon.grpc.interceptor.GrpcCallRateMeter;
+import monero.common.MoneroError;
+
+@Slf4j
+public class GrpcMoneroNodeService extends MoneroNodeImplBase {
+
+    private final CoreApi coreApi;
+    private final GrpcExceptionHandler exceptionHandler;
+
+    @Inject
+    public GrpcMoneroNodeService(CoreApi coreApi, GrpcExceptionHandler exceptionHandler) {
+        this.coreApi = coreApi;
+        this.exceptionHandler = exceptionHandler;
+    }
+
+    @Override
+    public void isMoneroNodeRunning(IsMoneroNodeRunningRequest request,
+                                    StreamObserver<IsMoneroNodeRunningReply> responseObserver) {
+        try {
+            var reply = IsMoneroNodeRunningReply.newBuilder()
+                    .setIsRunning(coreApi.isMoneroNodeRunning())
+                    .build();
+            responseObserver.onNext(reply);
+            responseObserver.onCompleted();
+        } catch (Throwable cause) {
+            exceptionHandler.handleException(log, cause, responseObserver);
+        }
+    }
+
+    @Override
+    public void getMoneroNodeSettings(GetMoneroNodeSettingsRequest request,
+                                      StreamObserver<GetMoneroNodeSettingsReply> responseObserver) {
+        try {
+            var reply = GetMoneroNodeSettingsReply.newBuilder()
+                    .setSettings(coreApi.getMoneroNodeSettings().toProtoMessage())
+                    .build();
```

Comment:
I get a NPE testing with new instances of arbitrator, Alice, and Bob for the first time. The error is resolved running them the second time.

In Alice's console:

```
java.lang.NullPointerException: null
	at bisq.daemon.grpc.GrpcMoneroNodeService.getMoneroNodeSettings(GrpcMoneroNodeService.java:84)
	at bisq.proto.grpc.MoneroNodeGrpc$MethodHandlers.invoke(MoneroNodeGrpc.java:419)
	at io.grpc.stub.ServerCalls$UnaryServerCallHandler$UnaryServerCallListener.onHalfClose(ServerCalls.java:172)
	at io.grpc.internal.ServerCallImpl$ServerStreamListenerImpl.halfClosed(ServerCallImpl.java:331)
	at io.grpc.internal.ServerImpl$JumpToApplicationThreadServerStreamListener$1HalfClosed.runInContext(ServerImpl.java:814)
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:123)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

Reply:
Thanks, fixed.